Sources/User.php

```diff
@@ -5370,7 +5370,7 @@ protected static function loadUserData(array $users, int $type = self::LOAD_BY_I
 				}
 
 				// Does the cached data have everything we need?
-				if (self::$dataset_levels[$data['dataset']] >= self::$dataset_levels[$dataset]) {
+				if (is_array($data) && !empty(self::$dataset_levels[$data['dataset']]) && !empty(self::$dataset_levels[$dataset]) && self::$dataset_levels[$data['dataset']] >= self::$dataset_levels[$dataset]) {
```

Comment:
The value of `self::$dataset_levels['minimal']` is zero, so this won't work correctly. 

Instead, I suggest a two-part solution.

First, on this line:

```suggestion
				if (is_array($data) && self::$dataset_levels[$data['dataset'] ?? 'minimal'] >= self::$dataset_levels[$dataset]) {
```

Then, at the top of this method, add the following in order to ensure that $dataset is valid:

```
		if (!isset(self::$dataset_levels[$dataset])) {
			$dataset = 'normal';
		}
```

Reply:
Yep, and I even noted it was an int and didn't think of that. I have applied your suggestion.